src/server/gc_worker/compaction_filter.rs

```diff
@@ -555,6 +580,36 @@ impl WriteCompactionFilter {
         if self.write_batch.count() > DEFAULT_DELETE_BATCH_COUNT || force {
             let err = match &mut self.write_batch.batch {
                 Either::Left(wb) => {
+                    // Acquire latch to prevent conflicts with ingestion operations
+                    // using RocksDB IngestExternalFileOptions.allow_write = true.
+                    let _region_inject_latch_guard = match self.engine.as_ref() {
+                        Some(engine) => {
+                            let smallest_key = self
+                                .write_batch
+                                .smallest_key
+                                .as_ref()
+                                .unwrap()
+                                .as_encoded()
+                                .clone();
+                            let largest_key = self
+                                .write_batch
+                                .largest_key
+                                .as_ref()
+                                .unwrap()
+                                .as_encoded()
+                                .clone();
+                            // Here, we can directly use [`smallest_key`, `largest_key`) for
+                            // locking. Since for the same key, the
+                            // `largest_key` here has a timestamp appended
+                            // at the end, while the locking for SST
+                            // ingestion uses `Region.start_key` without a
+                            // timestamp, an overlap exists, ensuring mutual exclusion.
+                            Some(engine.ingest_latch.acquire(smallest_key, largest_key))
```

Comment:
```suggestion
                            Some(engine.ingest_latch.acquire(smallest_key, keys::next_key(&largest_key)))
```
The upper bound of range should be larger than the biggest key

Reply:
Yes, this property is currently satisfied because the compaction filter uses keys with appended timestamps (e.g., bbb_11), while ingestion uses keys without timestamps (e.g., bbb). For example, if the compaction filter attempts to lock the range [aaa_1, bbb_11), it will be mutually exclusive with ingestion, which locks the range [bbb, ccc).

But, using keys::next_key is a simpler and more readable approach, so I will follow your advice.